frontend/src/Notifications/Notifications.jsx

```diff
@@ -18,8 +18,9 @@ export default function Notifications() {
         const fetchNotifications = async () => {
             try {
                 const response = await api.get('/api/user/notifications', {withCredentials: true});
-                setNotifications(response.data);
-                localStorage.setItem('notifications', JSON.stringify(response.data));
+                const filteredNotifications = response.data.filter(notif => notif.notificationsOn !== false);
```

Comment:
maybe i'm missing something, but should the notification itself have the attribute ``notificationsOn``? or does it belong to the patient that the notification is about?

Reply:
I decided to keep the notificationsOn attribute on the notification because using the notificationsOn attribute at the patient level would require more code and additional logic to ensure that all aspects of the application correctly respect the patient's notification preferences. Both methods work well tho.